canonical-web-and-design/maas-ui
# fix: machine list controls layout

## Done

- fix machine list controls across different screen sizes
  - use container queries which enables styles based on the size of the container and not just screen size (this allows for different styles based on the state of the side navigation)
  - replace `filter-accordion`, `filter-accordion__toggle` classes with a new `is-maas-select` class
  - add `renderButtons` prop to `SectionHeader`

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

-  Go to machine list
- Verify that action buttons group is displayed correctly across different screen sizes 
- Click an action to display the side panel
- verify everything works as before
- Go to machine details page
- Verify that action buttons group is displayed correctly across different screen sizes 
- Click an action to display the side panel
- verify everything works as before

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

Actions button group wrapping to a single next line when side navigation is expanded
![Kapture 2023-04-03 at 11 17 21](https://user-images.githubusercontent.com/7452681/229467017-5395ab37-93c5-4557-868b-4f27f6469cb1.gif)

Buttons get progressively wrapped as less space is available
<img width="1073" alt="image" src="https://user-images.githubusercontent.com/7452681/229467138-16384774-dcc3-40c5-b4a1-a56cb86abedd.png">

<img width="921" alt="image" src="https://user-images.githubusercontent.com/7452681/229467217-8522db2c-cfbd-4647-9a10-c63a0134f936.png">

<img width="846" alt="image" src="https://user-images.githubusercontent.com/7452681/229529979-f510af0a-b539-4cee-83b3-f6ba24a24d14.png">
